src/core_mymalloc.c

```diff
@@ -27,7 +26,7 @@ void *mymalloc(size_t n)
     }
 
     if(Nblocks >= MAXBLOCKS) {
-        printf("Nblocks = %lu No blocks left in mymalloc().\n", Nblocks);
+        printf("Nblocks = %ld No blocks left in mymalloc().\n", Nblocks);
         ABORT(OUT_OF_MEMBLOCKS);
```

Comment:
I think a lot of the `mymalloc` calls are threaded throughout the code.  Ideally we would adopt the scheme where we return `OUT_OF_MEMBLOCKS` and check at the caller level but I know this would be a massive pain to do.

Can refactoring this aspect be marked down for a future improvement?

Reply:
This whole memblocks things is a major PITA. There are other ways of tracking how much memory was consumed by the process. I would rather remove the entirely of `mymalloc` and do everything via `malloc`. But @darrencroton might want to weigh in on this